modules/transport-netty4/src/main/java/org/elasticsearch/transport/NettyAllocator.java

```diff
@@ -28,32 +28,59 @@
 import io.netty.channel.ServerChannel;
 import io.netty.channel.socket.nio.NioServerSocketChannel;
 import org.elasticsearch.common.Booleans;
+import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.monitor.jvm.JvmInfo;
 
 public class NettyAllocator {
 
     private static final ByteBufAllocator ALLOCATOR;
+    private static final String DESCRIPTION;
 
     private static final String USE_UNPOOLED = "es.use_unpooled_allocator";
     private static final String USE_NETTY_DEFAULT = "es.unsafe.use_netty_default_allocator";
+    private static final String USE_NETTY_DEFAULT_CHUNK = "es.unsafe.use_netty_default_chunk_and_page_size";
 
     static {
+        long g1gcRegionSizeInBytes = JvmInfo.jvmInfo().getG1RegionSize();
```

Comment:
I think this should check that we use g1 and if not, use region size -1. The variable in `JvmInfo` reflects the underlying `G1RegionsSize` as is and this seems to be 0 if using parallel GC. Safest to assume it "undefined" unless we use G1.